### PR TITLE
Pulls/usb hid cleanup

### PR DESCRIPTION
**Pull request by karlp**
please look at the individual commits for the details.

### CODE REVIEW

File: examples/stm32/f1/other/usb_hid/usbhid.c [hunk 1]
```diff
@@ -279,6 +279,13 @@ int main(void)
 	rcc_clock_setup_in_hsi_out_48mhz();
 
 	rcc_periph_clock_enable(RCC_GPIOA);
+	/* Hack to force renumeration */
+	gpio_set_mode(GPIOA, GPIO_MODE_OUTPUT_2_MHZ,
+		GPIO_CNF_OUTPUT_PUSHPULL, GPIO12);
+	gpio_clear(GPIOA, GPIO12);
+	for (unsigned i = 0; i < 800000; i++) {
```
Comment (fenugrec) [inline examples/stm32/f1/other/usb_hid/usbhid.c]: magic constant!

**Comment by fenugrec** [comment]
gpio A15 thing : what did that do, and who's going to fall in that trap ? maybe an explanatory note would be a good thing depending on the implications

HID example dupe removal : +1

File: examples/stm32/f1/other/usb_hid/usbhid.c (hunk 1 above)
Comment (karlp) [inline examples/stm32/f1/other/usb_hid/usbhid.c]: yep, it's pretty arbitrary how long you need to bounce for to trigger reenumeation.  we've got plenty of "hacky delays" like this in examples

**Comment by karlp** [comment]
A15 is JTDI, the old code was clearly for a board that was using a15 for some other purpose that wasn't stripped out completely when it was turned into the example and contributed.  I _suspect_ it was involved in vbus soft disconnect via an external fet, as most of this code had been submitted by gsmcmullin after developing the usb stack.  Total noise in general though.

**Comment by karlp** [comment]
merged after expanding the commentary on how the re-enumeration hack works.

**Pull request closed without merging**